agent/03_agent_deployment.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+LOGDIR=${SCRIPTDIR}/logs
+source $SCRIPTDIR/logging.sh
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/network.sh
+source $SCRIPTDIR/utils.sh
+source $SCRIPTDIR/validation.sh
+source $SCRIPTDIR/agent/common.sh
+
+early_deploy_validation
+
+function generate_fleeting_manifests() {
+
+    mkdir -p ${FLEETING_MANIFESTS_PATH}
+
+    if [[ "$IP_STACK" = "v4" ]]; then
+      CLUSTER_NETWORK=${CLUSTER_SUBNET_V4}
+      SERVICE_NETWORK=${SERVICE_SUBNET_V4}
+      CLUSTER_HOST_PREFIX=${CLUSTER_HOST_PREFIX_V4}
+    elif [[ "$IP_STACK" = "v6" ]]; then
+      CLUSTER_NETWORK=${CLUSTER_SUBNET_V6}
+      SERVICE_NETWORK=${SERVICE_SUBNET_V6}
+      CLUSTER_HOST_PREFIX=${CLUSTER_HOST_PREFIX_V6}
+    fi
+
+    cat > "${FLEETING_MANIFESTS_PATH}/agent-cluster-install.yaml" << EOF
+apiVersion: extensions.hive.openshift.io/v1beta1
+kind: AgentClusterInstall
+metadata:
+  name: test-agent-cluster-install
+  namespace: cluster0
+spec:
+  apiVIP: 192.168.111.10
+  ingressVIP: 192.168.111.11
+  clusterDeploymentRef:
+    name: compact-cluster
+  imageSetRef:
+    name: openshift-v4.10.0
```

Comment:
Is there a 4.11 nightly we could point to instead?

Reply:
IINW currently it's still hard-coded in fleeting, I was about to open a new PR for that

Reply:
you're right, @zaneb

Reply:
I don't have any objection to this being hard-coded for now, but in future IMO it would be better to derive the image version from the same variables we use to determine `OPENSHIFT_RELEASE_IMAGE`

In that case you have the option of getting latest (default), setting a specific `OPENSHIFT_RELEASE_IMAGE`, or asking for a specific version, e.g:

```
export OPENSHIFT_RELEASE_TYPE="ga"
export OPENSHIFT_VERSION="latest-4.10"
```

Reply:
TBH we could just ignore this field, since we're always going to use the version matching the installer binary.

Reply:
I think it's fine for the moment to ignore it, and review it in a later moment